src/v/CMakeLists.txt

```diff
@@ -37,23 +37,43 @@ add_subdirectory(compat)
 add_subdirectory(rp_util)
 
 option(ENABLE_GIT_VERSION "Build with Git metadata" OFF)
+
+# If enabled, the git hash is included in the Redpanda binary, which will
+# require relinking of at least the `redpanda` binary each time the hash
+# changes (switching branches, creating a new commit, etc).
+# Cannot be used when ENABLE_GIT_VERSION=off.
+option(ENABLE_GIT_HASH "Build with Git hash in metadata" OFF)
```

Comment:
To preserve behavior we'd set the default to `ON`, right?

I imagine the release builds depend on this, but not sure.

Reply:
Yeah, though the defaults will need to be set in our build taskfiles. You're right though that before that change is made, there will be a brief stint where builds will use ENABLE_GIT_VERSION=ON and ENABLE_GIT_HASH=OFF.

Reply:
PR for that here https://github.com/redpanda-data/vtools/pull/908